src/ir/proto/raksha_ir.proto

```diff
@@ -0,0 +1,101 @@
+//-----------------------------------------------------------------------------
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+
+// This proto file contains the proto representation of the Raksha IR. This
+// allows frontends to construct the Raksha IR in any language and send the IR
+// over the wire or across an FFI layer for analysis.
+
+syntax = "proto3";
+
+package raksha.ir.proto;
+
+message Module {
+  map<uint64, Block> blocks = 1;
```

Comment:
Why not just a repeated field?

Reply:
For the `Value` `BlockArgument`, we need to be able to specify the `Block` that the arguments are from. However, this `Value` shouldn't own the `Block`; the module that the `Block` resides in should own the `Block`. Therefore, we need some way to point at the `Block`. For that reason, I'm assigning an ID to each `Block`.

Of course, a `repeated` field can be thought of as a map from `index` to value, so we could provide the "map" that way. But I think it might be wise to not impose a density requirement unnecessarily, as that provides frontends the freedom to drop blocks. This might be useful, for instance, in the case of a TU containing errors where the frontend was still able to correctly parse some portions of the TU.

Reply:
Having an id sounds good. Is this going to be global id across all the blocks in the ir? Or, it is just scoped to the module? I assume the former? Add a brief comment like the one for `operations` in `Block`.

Reply:
To be honest, I wasn't 100% sure, but `uint64` should be enough for either. If you think an ID unique across the TU is the right call, that probably is the way to go. I'll add a comment explaining this.

Reply:
Done.